OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs

```diff
@@ -130,6 +131,32 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 					foreach (var s in buildingInfo.BuildSounds)
 						Game.Sound.PlayToPlayer(SoundType.World, order.Player, s, host.CenterPosition);
 				}
+				else if (os == "ReplaceBuilding")
+				{
+					if (!buildingInfo.IsCloseEnoughToBase(self.World, order.Player, order.TargetString, order.TargetLocation))
+						return;
+
+					// Current ReplaceBuilding-Order must contain ReplacementInfo
+					var replacementInfo = unit.TraitInfoOrDefault<ReplacementInfo>();
+					if (replacementInfo == null)
+						return;
+
+					// remove target Actor(s)
+					foreach (var t in buildingInfo.Tiles(order.TargetLocation))
+						if (self.World.WorldActor.Trait<BuildingInfluence>().GetBuildingAt(t) != null)
```

Comment:
We might want to cache `self.World.WorldActor.Trait<BuildingInfluence>()` in a local variable outside of the loop instead of querying it multiple times.